src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs

```diff
@@ -71,6 +71,7 @@ public struct AsyncTaskMethodBuilder<TResult>
             catch (Exception e)
             {
                 System.Threading.Tasks.Task.ThrowAsync(e, targetContext: null);
+                NullabilityHelpers.SuppressNonNullAssignmentWarning(ref taskField);
```

Comment:
Does it negatively affect the resulting asm to store the `GetStateMachineBox(ref stateMachine, ref taskField).MoveNextAction` into a local before the try block?  If it doesn't, that'd be my preferred solution.

Reply:
Or, actually, are any call sites relying on the [NotNull] annotation?  Even better would just be to delete that.